src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
```

Comment:
> SettingsDAO and mixxx.cfg don't seem any different in this regard, are they? You can store any QString / QByteArray in either of them. It's up to the code that reads/stores to it to parse/check. 

Before this PR SettingsDAO has serialized data which where likely misinterpreted after an update and mixxx.cfg has human readable and editable data. I like the idea to treat the data as "sacred", but we don't do this with the header state. 

What will be the rule of thump, which storage is used when?